src/useAsync.ts

```diff
@@ -18,14 +18,18 @@ export type AsyncState<T> =
       value: T;
     };
 
-const useAsync = <T>(fn: () => Promise<T>, deps: DependencyList = []) => {
-  const [state, callback] = useAsyncFn(fn, deps);
+export default function useAsync<
+  Result = any,
+  Args extends any[] = any[],
+  Fn extends Function = (...args: Args | []) => Promise<Result>
+>(fn: Fn, deps: DependencyList = []) {
+  const [state, callback] = useAsyncFn<Result, Args, Fn>(fn, deps, {
+    loading: true,
```

Comment:
this initial state here resolves the regression introduced by https://github.com/streamich/react-use/commit/4b998d6e0834e29f366574d668dc361c50be7ef3#diff-8e6e0808e6745d8058a2f9cbd476172dR22